spec/api.json

```diff
@@ -413,10 +413,57 @@
         },
         "type": "object"
       },
+      "Provider": {
+        "properties": {
+          "client_id": {
+            "description": "The RP's client identifier, issued by the IdP.",
+            "type": "string"
+          },
+          "config_url": {
+            "description": "A full path of the IdP config file.",
+            "type": "string"
+          },
+          "domain_hint": {
+            "description": "By specifying one of domain_hints values provided by the accounts endpoints,\nthe FedCM dialog selectively shows the specified account.",
+            "type": "string"
+          },
+          "fields": {
+            "description": "Array of strings that specifies the user information (\"name\", \" email\",\n\"picture\") that RP needs IdP to share with them.\n\nNote: Field API is supported by Chrome 132 and later.",
+            "items": {
+              "type": "string"
+            },
+            "type": "array"
+          },
+          "login_hint": {
+            "description": "By specifying one of login_hints values provided by the accounts endpoints,\nthe FedCM dialog selectively shows the specified account.",
+            "type": "string"
+          },
+          "nonce": {
+            "description": "A random string to ensure the response is issued for this specific request.\nPrevents replay attacks.",
+            "type": "string"
+          },
+          "parameters": {
+            "additionalProperties": {
+              "type": "string"
+            },
+            "description": "Custom object that allows to specify additional key-value parameters:\nscope: A string value containing additional permissions that RP needs to\nrequest, for example \" drive.readonly calendar.readonly\"\nnonce: A random string to ensure the response is issued for this specific\nrequest. Prevents replay attacks.\n\nOther custom key-value parameters.\n\nNote: parameters is supported from Chrome 132.",
+            "type": "object"
+          }
+        },
+        "type": "object"
+      },
       "RecoveryAddressType": {
         "title": "RecoveryAddressType must not exceed 16 characters as that is the limitation in the SQL Schema.",
         "type": "string"
       },
+      "SubmitFedcmTokenBody": {
```

Comment:
So far `submit` is not a keyword we use in the API spec. Is there an alternative that better aligns?

Reply:
A bit lengthy, but how about `createSessionFromFedCmToken`? We regularly use `create` elsewhere.

However, on registration, you really only get a session if the session hook is active, so the `createSession` part is really not correct in all cases.

Reply:
In a way `getFedcmParameters` starts the FedCM login/registration flow, and `submitFedcmToken` completes it, right? So maybe use the flow terminology instead: `createFedcmFlow` and `updateFedcmFlow`?